src/main/java/online/book/store/dto/BookSearchParametersDto.java

```diff
@@ -0,0 +1,14 @@
+package online.book.store.dto;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
+public record BookSearchParametersDto(
```

Comment:
I'm not sure, but if you want to search only by title and your author and isbn will be empty, it will cause of error. Give me feedback if I'm wrong, because I had the same problem.

Reply:
Yes, you're right, I didn't check that, probably, better not to use annotations in search parameters